### PR TITLE
feat(judicial-system): Retry upload on fail

### PR DESCRIPTION
**Pull request by oddsson**
# Retry upload on fail

https://app.asana.com/0/1199153462262248/1200123118127366

## What

If an upload fails for some reason, the user should be able to retry.

## Why

UX

## Screenshots / Gifs

https://user-images.githubusercontent.com/3789875/114197347-c4e1fa00-9941-11eb-84a9-1e9a5b950263.mov

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by barabrian — APPROVED** [review]
LGTM! could be nice for next iteration to add loading indicator on the retry, from a user perspective it kinda looks like nothing happened

File: apps/judicial-system/web/src/utils/useS3Upload.ts [hunk 1]
```diff
@@ -98,14 +108,8 @@ export const useS3Upload = (workingCase?: Case) => {
...
 
-    const indexOfFileInFiles = newFiles.findIndex(
-      (fileInFiles) => fileInFiles.name === file.name,
-    )
-
     const updatedFiles = newFiles.map((newFile) => {
-      return newFile.id === indexOfFileInFiles.toString()
-        ? Object.assign({}, newFile, { file })
-        : newFile
+      return newFile.id === file.id ? file : newFile
```
Comment (gudjong) [inline apps/judicial-system/web/src/utils/useS3Upload.ts:112]: Nice simplification

File: apps/judicial-system/web/src/utils/useS3Upload.ts [hunk 2]
```diff
@@ -187,9 +203,17 @@ export const useS3Upload = (workingCase?: Case) => {
...
+          setUploadErrorMessage(
+            'Upp kom óvænt kerfisvilla. Vinsamlegast reynið aftur.',
+          )
         })
     }
   }
 
-  return { files, onChange, onRemove }
+  const onRetry = (file: UploadFile) => {
+    setUploadErrorMessage(undefined)
```
Comment (gudjong) [inline apps/judicial-system/web/src/utils/useS3Upload.ts:214]: This is also done in onChange.

**Review by gudjong — APPROVED** [review]
Looks good.

**Pull request merged**